vividus-plugin-web-app-to-rest-api/src/main/java/org/vividus/steps/integration/ResourceCheckSteps.java

```diff
@@ -335,4 +347,21 @@ public void setUriToIgnoreRegex(Optional<String> uriToIgnoreRegex)
     {
         this.uriToIgnoreRegex = uriToIgnoreRegex;
     }
+
+    private boolean isInvalidResourcePresent(Set<WebPageResourceValidation> results)
+    {
+        for (WebPageResourceValidation resource : results)
+        {
+            if (resource.getCheckStatus() == CheckStatus.FAILED || resource.getCheckStatus() == CheckStatus.BROKEN)
+            {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private void recordFailedResourcesValidation()
+    {
+        softAssert.recordFailedAssertion("Found broken resources or resources with invalid status codes!");
+    }
```

Comment:
the same question: why do we need it? to trigger fail-fast action?

Reply:
> the same question: why do we need it? to trigger fail-fast action?

Yes
I also think about to use `AssertionFailedEvent` but it's seems not thread safety.